rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -157,6 +194,39 @@ void Player::play()
   }
 }
 
+void Player::pause()
+{
+  if (clock_) {
```

Comment:
How can `clock_` ever be `nullptr`? On that note, where is `clock_` currently initialized?

same below.

Reply:
Clock is currently intitialized in `prepare_clock`, which call from `play`. I would like to initialize it in the constructor (and therefore avoid having it be on the heap) - but we don't have access to the "initial time" until the reader is opened, which only happens when we call `play`.

It may make sense, to me, to open the reader in the constructor, now that we have all arguments present at that point. I wouldn't want to make that change here, though. What do you think?

Reply:
I am okay if you don't address it here, but I'd like to have it noted in an GH issue or an TODO to keep track of it. I think we should - now with its public header - review the overall API and offload some of the initialization work from `play` to the constructor.

Reply:
I agree with all of that